### ISSUE TITLE
Notify sources failed if invalid/not existed authentication_id get passed to the application_authentications API.

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#697]
In GitLab by @pakamble on May 15, 2020, 13:44

## Summary

Account creation with the sources API is invalid/not existed authentication_id get passed to the application_authentications API then error code and error message is not getting generated.

## Steps to Reproduce

1. start creating account with the sources API.
1. provide invalid/not existed value to the authentication_id in  application_authentications API call
1. eventually account creation is get failed.
1. check GET /application call and find value for the field "availability_status_error".

## Expected Result

- There should be proper error message and error code with the "availability_status_error" key.

## Actual Result

- No key found as name "availability_status_error" in GET /application call

## Additional context

Following are the API calls recorded while executing test case.

- http --verify=no --auth swatch-qe-test1:shadowman --body post https://qa.cloud.redhat.com/api/sources/v1.0/sources/ name=cloudigrade_automation source_type_id=2
SOURCEID :: 75807

- http --verify=no --auth swatch-qe-test1:shadowman --body post https://qa.cloud.redhat.com/api/sources/v1.0/endpoints/ role=aws default=true source_id=75807
ENDPOINT:: 48792

- http --verify=no --auth swatch-qe-test1:shadowman --body post https://qa.cloud.redhat.com/api/sources/v1.0/authentications/ authtype=cloud-meter-arn password=arn:aws:iam::439727791560:role/CloudigradeRoleForTestEnv resource_type=ENDPOINT resource_id=48792

- http --verify=no --auth swatch-qe-test1:shadowman --body post https://qa.cloud.redhat.com/api/sources/v1.0/applications/ application_type_id=5 source_id=75807
APPLICATIONID :: 55891

- http --verify=no --auth swatch-qe-test1:shadowman GET https://qa.cloud.redhat.com/api/sources/v1.0/applications/55891/

```
HTTP/1.1 200 OK
X-Powered-By: Express
cache-control: max-age=0, private, must-revalidate
connection: keep-alive
content-type: application/json; charset=utf-8
date: Fri, 15 May 2020 16:11:41 GMT
etag: W/"b37dae3fe784e7e9ecdeedbaf24d85aa"
server: openresty/1.13.6.1
set-cookie: a549c90418df47840900855e57bbdfae=9af2f5a78c06a3a3c5ccefeb80ec1065; path=/; HttpOnly; Secure
transfer-encoding: chunked
vary: Origin
x-content-type-options: nosniff
x-fakamai-app-hint: API
x-fakamai-downstream-host: https://api-gateway-qa.5a9f.insights-dev.openshiftapps.com
x-fakamai-env: qa
x-request-id: 6ddc92b3-2a26-4093-953a-8bb5c5799053
x-rh-insights-request-id: 02f812d175174bca8a0ae615c67bcc3a
x-runtime: 0.007301

{
    "application_type_id": "5",
    "created_at": "2020-05-15T16:11:39Z",
    "id": "55891",
    "source_id": "75807",
    "tenant": "6675444",
    "updated_at": "2020-05-15T16:11:39Z"
}
```

- http --verify=no --auth swatch-qe-test1:shadowman --body post https://qa.cloud.redhat.com/api/sources/v2.0/application_authentications/ application_id=55891 authentication_id=123333
APPLICATION_AUTHETICATIO_ID :: null

- http --verify=no --auth swatch-qe-test1:shadowman https://qa.cloud.redhat.com/api/cloudigrade/v2/accounts/
 
```
HTTP/1.1 200 OK
Transfer-Encoding: chunked
X-Powered-By: Express
allow: GET, POST, HEAD, OPTIONS
cache-control: private
connection: close
content-type: application/json
date: Fri, 15 May 2020 16:11:55 GMT
server: openresty/1.13.6.1
set-cookie: a549c90418df47840900855e57bbdfae=9af2f5a78c06a3a3c5ccefeb80ec1065; path=/; HttpOnly; Secure
vary: Accept
x-cloudigrade-request-id: 572771ab-93d5-495e-89ac-ecfff33ebecb
x-content-type-options: nosniff
x-fakamai-app-hint: API
x-fakamai-downstream-host: https://api-gateway-qa.5a9f.insights-dev.openshiftapps.com
x-fakamai-env: qa
x-frame-options: DENY
x-rh-insights-request-id: 5fc4cc5ba31042988b10b075f7051df4

{
    "data": [],
    "links": {
        "first": "/api/cloudigrade/api/cloudigrade/v2/accounts/?limit=10&offset=0",
        "last": "/api/cloudigrade/api/cloudigrade/v2/accounts/?limit=10&offset=0",
        "next": null,
        "previous": null
    },
    "meta": {
        "count": 0
    }
}
```

**Comment by katherine-black** [comment]
In GitLab by @pakamble on May 18, 2020, 05:58

Closing this issue as this is happen due to wrong test step. 

following comment made by @bizhang with specific reason. 

I looked through the logs you sent me, the cause of both is that the sources Application_Authentication doesn't get created in Sources so there's nothing to do on the cloud meter side. Sources doesn't let you create and endpoint with invalid resource_type, nor does it let you create an app_auth with an invalid auth_id. Because the App_Auth doesn't get created, the cloud meter listener never gets the message it needs to tell it to create the cloud account.
